src/jose/jws/general/verifier.ts

```diff
@@ -11,22 +13,33 @@ type VerificationResult = {
   signers: string[];
 };
 
-// TODO: add logic to prevent validating duplicate signatures, Issue #66 https://github.com/TBD54566975/dwn-sdk-js/issues/66
 export class GeneralJwsVerifier {
   jws: GeneralJws;
+  cache: Cache;
 
-  constructor(jws: GeneralJws) {
+  constructor(jws: GeneralJws, cache?: Cache) {
     this.jws = jws;
+    this.cache = cache || new MemoryCache(600);
   }
 
   async verify(didResolver: DidResolver): Promise<VerificationResult> {
     const signers: string[] = [];
 
     for (const signatureEntry of this.jws.signatures) {
+      let isVerified: boolean;
+      const cacheKey = `${signatureEntry.protected}.${this.jws.payload}.${signatureEntry.signature}`;
       const kid = GeneralJwsVerifier.getKid(signatureEntry);
       const publicJwk = await GeneralJwsVerifier.getPublicKey(kid, didResolver);
 
-      const isVerified = await GeneralJwsVerifier.verifySignature(this.jws.payload, signatureEntry, publicJwk);
+      const cachedValue = await this.cache.get(cacheKey);
+
+      if (cachedValue === undefined) {
+        isVerified = await GeneralJwsVerifier.verifySignature(this.jws.payload, signatureEntry, publicJwk);
+        await this.cache.set(cacheKey, isVerified.toString());
```

Comment:
```suggestion
        await this.cache.set(cacheKey, isVerified);
```

Reply:
If the Cache provider returns serialized values, then even if we cache `false`, it would return from cache as `'false'` which would be truthy, no? so we wouldn't reverify the signature, but `if(isVerified)` would pass and the signature would get added to the list of signers?

Reply:
The cache shouldn't  morph the data cached, but valid concern, I'd say we can put the `===` on the `if(isVerified)`:

`if (isVerified === true) {`

And add comments above why strict equality is there as it is not immediately obvious. What do you think?